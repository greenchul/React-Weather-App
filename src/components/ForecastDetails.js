import React from "react";
import propTypes from "prop-types";
import moment from "moment";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecastDetail }) => {
  const { date, temperature, wind, humidity } = forecastDetail;
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__max-temp">
        Max Temperature: {temperature.max} &deg;C
      </div>
      <div className="forecast-details__min-temp">
        Min Temperature: {temperature.min} &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__wind">
        Wind: {wind.speed}mph {wind.direction}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecastDetail: propTypes.shape({
    date: propTypes.number.isRequired,
    temperature: propTypes.shape({
      max: propTypes.number,
      min: propTypes.number,
    }).isRequired,
    wind: propTypes.shape({
      speed: propTypes.number.isRequired,
      direction: propTypes.string.isRequired,
    }),
    humidity: propTypes.number.isRequired,
  }).isRequired,
};
export default ForecastDetails;
