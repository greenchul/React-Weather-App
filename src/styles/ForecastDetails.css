.forecast-details {
  padding-bottom: 20px;
  padding-top: 10px;
  font-size: 20px;
}

.forecast-details__date {
  font-weight: 700;
}

@media screen and (min-width: 768px) {
  .forecast-details {
    font-size: 25px;
    line-height: 35px;
  }
}

@media screen and (min-width: 1024px) {
  .forecast-details {
    display: flex;
    justify-content: space-between;
  }
}
