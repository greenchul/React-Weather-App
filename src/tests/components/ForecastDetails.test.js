import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails component", () => {
  const validProps = {
    date: 1525046400000,
    temperature: {
      max: 23,
      min: 15,
    },
    humidity: 89,
    wind: {
      speed: 12,
      direction: "N",
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails forecastDetail={validProps} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct date", () => {
    render(<ForecastDetails forecastDetail={validProps} />);
    const dateElement = screen.getByText(/Mon 30th Apr/i);

    expect(dateElement).toBeInTheDocument();
    expect(dateElement).toHaveClass("forecast-details__date");
  });

  it("renders correct max temperature", () => {
    render(<ForecastDetails forecastDetail={validProps} />);
    const maxTempElement = screen.getByText(/23/i);

    expect(maxTempElement).toBeInTheDocument();
    expect(maxTempElement).toHaveClass("forecast-details__max-temp");
  });

  it("renders correct min temperature", () => {
    render(<ForecastDetails forecastDetail={validProps} />);
    const minTempElement = screen.getByText(/15/);

    expect(minTempElement).toBeInTheDocument();
    expect(minTempElement).toHaveClass("forecast-details__min-temp");
  });

  it("renders correct humidity", () => {
    render(<ForecastDetails forecastDetail={validProps} />);
    const humidityElement = screen.getByText(/89%/i);

    expect(humidityElement).toBeInTheDocument();
    expect(humidityElement).toHaveClass("forecast-details__humidity");
  });

  it("renders correct wind", () => {
    render(<ForecastDetails forecastDetail={validProps} />);
    const windElement = screen.getByText(/12mph N/i);

    expect(windElement).toBeInTheDocument();
    expect(windElement).toHaveClass("forecast-details__wind");
  });
});
