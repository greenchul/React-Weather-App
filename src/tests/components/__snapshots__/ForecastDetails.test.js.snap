// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon 30th Apr
    </div>
    <div
      class="forecast-details__max-temp"
    >
      Max Temperature: 23 °C
    </div>
    <div
      class="forecast-details__min-temp"
    >
      Min Temperature: 15 °C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 89%
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 12mph N
    </div>
  </div>
</DocumentFragment>
`;
